packages/x-tree-view/src/TreeItem/TreeItem.tsx

```diff
@@ -208,6 +210,8 @@ export const TreeItem = React.forwardRef(function TreeItem(
     ...other
   } = props;
 
+  const { handleExpansion } = useTreeItemState(itemId);
```

Comment:
This feels weird to me, because of the way we get the status values below. 🙈 And I didn't want to re-write this method so I took it from `useTreeItemState`. I'd love to know if there is a better way for doing this 😬 
PS. `TreeItem2` is the best 💙

Reply:
We could use `useTreeItemState` to get the status now that we need to use this hook here :+1: 
I think it's the best approach we have.